tasks/trace_agent.py

```diff
@@ -83,3 +83,28 @@ def integration_tests(ctx, install_deps=False, race=False, remote_docker=False):
 
     for prefix in prefixes:
         ctx.run("{} {}".format(go_cmd, prefix))
+
+@task
+def cross_compile(ctx, tag=""):
+    """
+    Cross-compiles the trace-agent binaries. Use the "--tag=X" argument to specify build tag.
+    """
+    if tag == "":
+        print("Argument --tag=<version> is required.")
+        return
+
+    print("Buliding tag %s..." % tag)
+
+    args = { "tag": tag }
+    env = { "TRACE_AGENT_VERSION": tag }
+
+    ctx.run("git checkout {tag}".format(**args))
+    ctx.run("mkdir -p ./bin/trace-agent/{tag}".format(**args))
+    ctx.run("go generate ./pkg/trace/info", env=env)
+    ctx.run("go get -u github.com/karalabe/xgo")
+    ctx.run("xgo -dest=bin/trace-agent/{tag} -go=1.11 -out=trace-agent-{tag} -targets=windows-6.1/amd64,linux/amd64,darwin-10.11/amd64 ./cmd/trace-agent".format(**args))
+    ctx.run("mv ./bin/trace-agent/{tag}/trace-agent-{tag}-windows-6.1-amd64.exe ./bin/trace-agent/{tag}/trace-agent-{tag}-windows-amd64.exe".format(**args))
```

Comment:
Even for the lines that have multiple `{tag}`, you can just do `.format(tag=tag)` - no need for unpacking of `args` unless you have more than one key.

Although if you expect `args` to eventually have more keys, then what you have makes sense as a more general solution.

Reply:
You're right, I've completely removed both `args` and `env`.